app/main.py

```diff
@@ -289,6 +314,23 @@ def update_live_alerts_data(
     # And we deduce the subset of alerts that we can deem to be "live"
     live_alerts = all_alerts[mask_acknowledgement].copy()
 
+    # We then fetch sunrise and sunset times and add a safety margin of 30 min (converting from UTC) to cover night time
+    sunrise = night_time_data['results']['sunrise'][:-6]
+    sunrise = datetime.fromisoformat(str(sunrise)) + timedelta(hours=2.5)
+    sunrise = sunrise.time()
+
+    sunset = night_time_data['results']['sunset'][:-6]
+    sunset = datetime.fromisoformat(str(sunset)) + timedelta(hours=1.5)
+    sunset = sunset.time()
+
+    # Are there some live alerts during night time ? If yes let's filter them out
```

Comment:
I would rather tend to use a `.map` and boolean indexing here. I am not a specialist of performance matters but it seems to me than it is generally faster than writing `for` loops. This could give something like: 

```
mask = live_alerts['created_at'].map(lambda x: is_hour_between(sunrise, sunset, x))
live_alerts = live_alerts[mask].copy()
```

Let me know what you think about this 🙂

Reply:
Thanks for this indeed it seems to be more efficient !